clicommand/cache_restore.go

```diff
@@ -0,0 +1,98 @@
+package clicommand
+
+import (
+	"context"
+	"slices"
+
+	"github.com/buildkite/agent/v3/internal/cache"
+	"github.com/urfave/cli"
+)
+
+const cacheRestoreHelpDescription = `Usage:
+
+    buildkite-agent cache restore [options]
+
+Description:
+
+Restores files from the cache for the current build based on the cache configuration
+defined in your cache config file (defaults to .buildkite/cache.yml).
+
+The cache configuration file defines which files or directories should be restored
+and their associated cache keys. Caches are scoped by organization, pipeline, and
+branch. If an exact cache match is not found, the command will attempt to use
+fallback keys if defined in your cache configuration.
+
+Note: This feature is currently in development and subject to change. It is not
+yet available to all customers.
+
+Example:
+
+    $ buildkite-agent cache restore
+
+This will restore all caches defined in .buildkite/cache.yml. You can also restore
+specific caches by providing their IDs:
+
+    $ buildkite-agent cache restore --ids "node"
+
+The cache will be retrieved from the bucket specified by --bucket-url or your
+cache configuration.
+
+Configuration File Format:
+
+The cache configuration file should be in YAML format:
+
+    - id: node
+      key: '{{ id }}-{{ agent.os }}-{{ agent.arch }}-{{ checksum "package-lock.json" }}'
+	  fallback_keys:
+		- '{{ id }}-{{ agent.os }}-{{ agent.arch }}-'
+      paths:
+        - node_modules
```

Comment:
It might be wise to we document the behavior when it comes to symlink in the folder.

Reply:
@zhming0 yeah this will need to be captured as it also overlaps with security, which makes it very complicated.